lib/go-tc/enum.go

```diff
@@ -191,7 +191,7 @@ type DeepCachingType string
 const (
 	DeepCachingTypeAlways  = DeepCachingType("ALWAYS")
 	DeepCachingTypeNever   = DeepCachingType("NEVER")
-	DeepCachingTypeInvalid = DeepCachingType("")
+	DeepCachingTypeInvalid = DeepCachingType("INVALID")
```

Comment:
The default value of the enum should match the default value of the type, i.e. `""`.  So per 
```
case "":
+		// default when omitted
+		return DeepCachingTypeNever
``` 
below, `DeepCachingTypeNever   = DeepCachingType("NEVER")` should be 
`DeepCachingTypeNever   = DeepCachingType("")`

This will prevent errors if someone default-initializes a `DeepCachingType`, e.g. `t := DeepCachingType{}` or `t := DeepCachingType("")` expecting it to be the default for the enum.

Reply:
good point..   I'll change that..

Reply:
Is this ever serialised to JSON, e.g. for the Server? Might need
`func (t *DeepCachingType) MarshalJSON() ([]byte, error) { return []byte(t.String()), nil }` and to change `func (t DeepCachingType) String()` to `case DeepCachingTypeNever:
return "NEVER"`, too.

That will make everything but `string(t)` work (which should never be done). There's no perfect solution, but this is probably the safest, and is the only way to make `DeliveryService{}` default to Never.